interfaces/inc/file.h

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#ifndef FILE_H
+#define FILE_H
+
+#ifdef __cplusplus
+#include <cstdint>
+#else
+#include <stdbool.h>
+#include <stdint.h>
+#endif
+
+#include "umock_c/umock_c_prod.h"
+#include "azure_macro_utils/macro_utils.h"
+#include "constbuffer.h"
+#include "execution_engine.h"
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#define FILE_WRITE_ASYNC_VALUES \
+    FILE_WRITE_ASYNC_INVALID_ARGS, \
+    FILE_WRITE_ASYNC_WRITE_ERROR, \
+    FILE_WRITE_ASYNC_ERROR,\
+    FILE_WRITE_ASYNC_OK
+MU_DEFINE_ENUM(FILE_WRITE_ASYNC_RESULT, FILE_WRITE_ASYNC_VALUES);
+
+#define FILE_READ_ASYNC_VALUES \
+    FILE_READ_ASYNC_INVALID_ARGS, \
+    FILE_READ_ASYNC_READ_ERROR, \
+    FILE_READ_ASYNC_ERROR,\
+    FILE_READ_ASYNC_OK
+MU_DEFINE_ENUM(FILE_READ_ASYNC_RESULT, FILE_READ_ASYNC_VALUES);
+
+typedef void *FILE_HANDLE;
+typedef void(*FILE_WRITE_CB)(void* user_context, bool is_successful);

```

Comment:
>bool [](start = 49, length = 4)

Since you created the enums for the return value, we should use them for the callbacks also.

Reply:
Should the read and write callbacks have the same enum?

---
In reply to: [448498837](https://github.com/Azure/azure-c-pal/pull/7#discussion_r448498837) [](ancestors = 448498837)